src/Features/Inventoriable.php

```diff
@@ -245,50 +245,63 @@ public function getInventoryAgent(): ?Agent
     {
         global $DB;
 
-        $agent = new Agent();
-        $iterator = $DB->request([
-            'SELECT'    => ['id'],
-            'FROM'      => Agent::getTable(),
-            'WHERE'     => [
-                'itemtype' => $this->getType(),
-                'items_id' => $this->fields['id']
-            ],
-            'ORDERBY'   => ['last_contact DESC'],
-            'LIMIT'     => 1
-        ]);
+        $get_most_recent_agent = static function (array $conditions) use ($DB): ?Agent {
```

Comment:
Why an inline function rather than another method (that should potentially be tested)?

Reply:
I refactored this part and add some tests.